src/field/launch_application.c

```diff
@@ -0,0 +1,1400 @@
+    Unk0203F134 *data = TaskManager_GetEnvironment(taskman);
+
+    switch (data->state) {
+    case 0:
+        sub_0203F1E8(fieldSystem, data->unkC);
+        data->state++;
+        break;
+    case 1:
+        if (!FieldSystem_ApplicationIsRunning(fieldSystem)) {
+            u16 *ptr = data->unk4;
+            if (ptr != NULL) {
+                if (data->unk8 == 1) {
+                    *ptr = 1;
+                } else {
+                    *ptr = 0;
+                }
+            }
+            FreeToHeap(data->unkC);
+            FreeToHeap(data);
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+void sub_0203F198(TaskManager *taskman, u16 *ret, SaveData *saveData, u16 a3, u16 a4) {
+    u32 size;
+    u8 *ptr;
+
+    Unk0203F134 *data = AllocFromHeap(HEAP_ID_32, sizeof(Unk0203F134));
+
+    ptr = (u8 *)data;
+    size = sizeof(Unk0203F134);
+    while (size) {
+        *ptr = 0;
+        ptr++;
+        size--;
+    }
+    data->unkC = sub_0203F0D0(HEAP_ID_32, saveData, a3, &data->unk8, a4);
+    data->unk4 = ret;
+    TaskManager_Call(taskman, sub_0203F134, data);
+}
+
+static BOOL sub_0203F1E8(FieldSystem *fieldSystem, Unk0203F0D0 *args) {
+    args->unk20 = &fieldSystem->unk_10C;
+    FieldSystem_LaunchApplication(fieldSystem, &_0210F9AC, args);
+    return TRUE;
+}
+
+BOOL LaunchAccessoryPortraitApp(FieldSystem *fieldSystem, FashionAppData *args) {
+    FieldSystem_LaunchApplication(fieldSystem, &sOvyTemplate_AccessoryPortrait, args);
+    return TRUE;
+}
+
+static void InitWirelessTradeSelectMonArgs(WirelessTradeSelectMonArgs *args, FieldSystem *fieldSystem) {
+    args->profile = Save_PlayerData_GetProfileAddr(fieldSystem->saveData);
+    args->party = SaveArray_Party_Get(fieldSystem->saveData);
+    args->palPad = SaveArray_Get(fieldSystem->saveData, SAVE_PALPAD);
+    args->unk14 = sub_0202CA44(fieldSystem->saveData);
+    args->options = Save_PlayerData_GetOptionsAddr(fieldSystem->saveData);
+    args->pokedex = Save_Pokedex_Get(fieldSystem->saveData);
+    args->natDexEnabled = SaveArray_IsNatDexEnabled(fieldSystem->saveData);
+    args->saveData = fieldSystem->saveData;
+    args->gameStats = Save_GameStats_Get(fieldSystem->saveData);
+    args->partnerProfile = AllocFromHeap(HEAP_ID_32, PlayerProfile_sizeof());
+    args->unk38 = AllocFromHeap(HEAP_ID_32, sub_02070D90());
+    args->unk3C = AllocFromHeap(HEAP_ID_32, sub_02070D90());
+    args->fieldSystem = fieldSystem;
+    args->unk30 = 0;
+}
+
+static void FreeWirelessTradeSelectMonArgs(WirelessTradeSelectMonArgs *args) {
+    if (args->partnerProfile != NULL) {
+        FreeToHeap(args->partnerProfile);
+        args->partnerProfile = NULL;
+    }
+    if (args->unk38 != NULL) {
+        FreeToHeap(args->unk38);
+        args->unk38 = NULL;
+    }
+    if (args->unk3C != NULL) {
+        FreeToHeap(args->unk3C);
+        args->unk3C = NULL;
+    }
+}
+
+static BOOL _Task_WirelessTrade(TaskManager *taskman) {
```

Comment:
check whether this is actually a task or launches one

Reply:
This is a task, renamed to `Task_WirelessTrade`.